movement/watch_faces/complication/day_one_face.c

```diff
@@ -85,96 +113,143 @@ bool day_one_face_loop(movement_event_t event, movement_settings_t *settings, vo
+            }
             break;
         case EVENT_LIGHT_BUTTON_UP:
             // otherwise use the light button to advance settings pages.
-            if (state->current_page != 0) {
-                // go to next setting page...
-                state->current_page = (state->current_page + 1) % 4;
-                if (state->current_page == 0) {
-                    // ...unless we've been pushed back to display mode.
-                    movement_request_tick_frequency(1);
-                    // force display since it normally won't update til midnight.
-                    _day_one_face_update(*state);
-                }
+            switch (state->current_page) {
+                case PAGE_YEAR:
+                    // fall through
+                case PAGE_MONTH:
+                    // fall through
+                case PAGE_DAY:
+                    // go to next setting page...
+                    state->current_page = (state->current_page + 1) % 4;
+                    if (state->current_page == PAGE_DISPLAY) {
+                        // ...unless we've been pushed back to display mode.
+                        movement_request_tick_frequency(1);
+                        // force display since it normally won't update til midnight.
+                        _day_one_face_update(state);
+                    }
+                    break;
+                default:
+                    break;
             }
             break;
         case EVENT_ALARM_BUTTON_UP:
             // if we are on a settings page, increment whatever value we're setting.
-            if (state->current_page != 0) {
-                state->birthday_changed = true;
-                switch (state->current_page) {
-                    case 1:
-                        state->birth_year = state->birth_year + 1;
-                        if (state->birth_year > state->current_year) state->birth_year = 1900;
-                        break;
-                    case 2:
-                        state->birth_month = (state->birth_month % 12) + 1;
-                        break;
-                    case 3:
-                        state->birth_day = state->birth_day + 1;
-                        if (state->birth_day == 0 || state->birth_day > days_in_month[state->birth_month - 1]) {
-                            state->birth_day = 1;
-                        }
-                        break;
-                }
+            switch (state->current_page) {
+                case PAGE_YEAR:
+                    // fall through
+                case PAGE_MONTH:
+                    // fall through
+                case PAGE_DAY:
+                    _day_one_face_abort_quick_cycle(state);
+                    _day_one_face_increment(state);
+                    break;
+                case PAGE_DISPLAY:
+                    state->current_page = PAGE_DATE;
+                    sprintf(buf, "%04d%02d%02d", state->birth_year, state->birth_month, state->birth_day);
```

Comment:
This triggers a (somewhat spurious) warning about a possible overflow. While it won't happen, could you still change this to something like:

```c
sprintf(buf, "%04d%02d%02d", state->birth_year % 10000, state->birth_month % 100 , state->birth_day % 100);
```

to silence it?